src/bcccp/carpark/exit/ExitController.java

```diff
@@ -6,149 +6,320 @@
 import bcccp.carpark.ICarpark;
 import bcccp.carpark.IGate;
 import bcccp.tickets.adhoc.IAdhocTicket;
+import bcccp.carpark.ICarSensor;
+import bcccp.carpark.ICarSensorResponder;
+import bcccp.carpark.ICarpark;
+import bcccp.carpark.IGate;
+import bcccp.tickets.adhoc.IAdhocTicket;
 import java.util.*;
+
 public class ExitController 
-		implements ICarSensorResponder,
-		           IExitController {
-	enum State
-    {
-		IDLE, WAITING, BLOCKED, PROCESSED, REJECTED, TAKEN, EXITING, EXITED
-    }
+implements ICarSensorResponder,
+           IExitController {
+
+private enum STATE { IDLE, WAITING, PROCESSED, REJECTED, TAKEN, EXITING, EXITED, BLOCKED } 
+
+private STATE state;
+private STATE prevState;
+private String message;
```

Comment:
maybe change the private variable names according to convention of underscore?

```
private STATE state_;
private STATE prevState_;
private String message_;
```

Reply:
yes try that one please to resolve the error of private variable